extension/src/ALObject/ALControl.ts

```diff
@@ -0,0 +1,265 @@
+import { ALControlType, ALObjectType, ALPropertyType, MultiLanguageType, XliffTokenType } from "./Enums";
+import { MultiLanguageObject } from "./MultiLanguageObject";
+import { XliffIdToken } from "./XliffIdToken";
+
+export class ALControl extends ALElement {
+    type: ALControlType = ALControlType.None;
+    _name?: string;
+    private _value?: string;
+    xliffTokenType: XliffTokenType = XliffTokenType.InheritFromControl;
+    multiLanguageObjects: MultiLanguageObject[] = new Array();
+    controls: ALControl[] = new Array();
+    properties: ALProperty[] = new Array();
+    isALCode: boolean = false;
+
+    constructor(type: ALControlType, name?: string, value?: string) {
+        super();
+        this.type = type;
+        if (name) {
+            this.name = name;
+        }
+        if (value) {
+            this.value = value;
+        }
+    }
+
+    public get name(): string {
+        if (!this._name) {
+            return '';
+        }
+        return this._name;
+    }
+    public set name(name: string) {
+        name = name.trim();
+        if (name.toLowerCase().startsWith('rec.')) {
+            name = name.substr(4);
+        }
+        this._name = Common.TrimAndRemoveQuotes(name);
+    }
+
+    public get value(): string {
+        if (!this._value) {
+            return '';
+        }
+        return this._value;
+    }
+    public set value(value: string) {
+        value = value.trim();
+        if (value.toLowerCase().startsWith('rec.')) {
+            value = value.substr(4);
+        }
+        this._value = Common.TrimAndRemoveQuotes(value);
+    }
+
+
+    public get caption(): string {
+        let prop = this.multiLanguageObjects.filter(x => x.name === MultiLanguageType[MultiLanguageType.Caption])[0];
+        if (!prop) {
+            let object = this.getObject();
+            if ([ALObjectType.Page, ALObjectType.PageExtension].includes(object.objectType) && [ALControlType.PageField, ALControlType.Part].includes(this.type)) {
+                // Check table for caption
+                let objects = this.getAllObjects();
+
+                if (objects) {
+                    if (this.type === ALControlType.Part) {
+                        let part = <unknown>this;
+                        let castedPart = <ALPagePart>part; // Workaround since "(this instanceof ALPagePart)" fails with "TypeError: Class extends value undefined is not a constructor or null"
+                        let relatedObj = castedPart.relatedObject;
+                        if (relatedObj) {
+                            return relatedObj?.caption;
+                        }
+                    } else {
+                        let sourceObject;
+                        if (object.objectType === ALObjectType.Page) {
+                            if (object.sourceTable !== '') {
+                                sourceObject = objects.filter(x => (x.objectType === ALObjectType.Table && x.name === object.sourceTable))[0]; // TODO:
```

Comment:
Do you have something left TODO here? :)

Reply:
don't remember :D